src/src/main/java/factories/ProductServiceFactory.java

```diff
@@ -0,0 +1,18 @@
+package factories;
+
+import service.Impl.ProductServiceImpl;
+import service.ProductService;
+
+public class ProductServiceFactory {
+    private static ProductService instance;
+
+    private ProductServiceFactory() {
+    }
+
+    public static synchronized ProductService getInstance() {
```

Comment:
why we use word 'synchronized' in this class and don't use in class UserDAOFactory?